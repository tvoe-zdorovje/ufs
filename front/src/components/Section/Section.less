@import "../Shared";

.section {
  margin-top: 12px;
  border: 1px solid @green;
}

.header {
  padding: 1px 8px 1px 4px;
  margin: -12px 0 0 8px;
  color: @green;
  background-color: #fff;
  position: absolute;
  z-index: 10;
}

.container {
  padding: 10px 5px 5px 5px;
}
