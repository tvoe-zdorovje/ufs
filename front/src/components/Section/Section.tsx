import React from 'react'

const styles = require('./Section.less')

interface SectionProps {
  header: string,
  children?: any,
}

export default (props: SectionProps) => (
  <section className={styles.section}>
    <div className={styles.header}>{props.header}</div>
    <div className={styles.container}>
      {props.children}
    </div>
  </section>
);
