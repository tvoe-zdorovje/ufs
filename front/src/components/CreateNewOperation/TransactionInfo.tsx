import React from 'react'
import {
  Form,
  Col,
  Input,
  Row,
  Button,
  Modal,
} from 'antd'
const FormItem = Form.Item;

import Section from 'components/Section/Section'
import ClientCard from  './ClientCard'
import { Task } from 'interfaces/dto/Task.interface'
import { Representative } from 'interfaces/dto/Representative.interface'
import m2w from 'utils/MoneyToWords'

const styles = require('./CreateNewOperation.less')

export interface TransactionInfoProps {
  getFieldDecorator: Function,
  representative: Representative,
  task: Task,
  ovnBased?: boolean,
  readOnly?: boolean,
}

export default class TransactionInfo extends React.Component<TransactionInfoProps, any> {
  state = { visible: false }
  toggleModal = () => {
    this.setState({
      visible: !this.state.visible,
    });
  }
  handleCancel = (e) => {
    this.setState({
      visible: false,
    });
  }

  render() {
    const { getFieldDecorator, readOnly = false, ovnBased = false, representative } = this.props
    const task = { ...this.props.task, amountInWords: m2w(this.props.task.amount) }
    const disabledInput = (key) => (<FormItem>
      {(typeof task[key] === 'undefined') ?
        <Input disabled />
        :
        <Input disabled value={task[key]} />}
    </FormItem>)

    const requiredInput = (key, emptyMessage) => {
      if (readOnly) {
        return disabledInput(key)
      }
      let config = { rules: [{ required: true, message: emptyMessage }] }
      if (typeof task[key] !== 'undefined') {
        config = { ...config, initialValue: task[key] }
      }
      return (
        <FormItem>
          {getFieldDecorator(key, config)(
            <Input />
          )}
        </FormItem>
      )
    }

    const amountInput = () => {
      const key = 'amount'
      if (readOnly || ovnBased) {
        return (<Input disabled value={`${task.amount} 00 коп.`} />)
      }
      let config = {
        rules: [{ required: true, message: 'Укажите сумму операции' },
          { pattern: (new RegExp('^\\d+$')), message: 'сумма операции должна быть целым числом' }],
      }
      if (typeof task.amount !== 'undefined') {
        config = { ...config, initialValue: task.amount }
      }
      return (
        <FormItem>
          {getFieldDecorator(key, config)(
            <Input addonAfter=" 00 коп." />
          )}
        </FormItem>
      )
    }

    const label = (value) => (<span className={styles.label}>{value}</span>)
    return (
      <div className="transactionCard">
        <Section header="Основная информация">
          <Row>
            <Col span={24}>
              <Row>
                <Col span={10}>
                  {label('Объявление №')}
                  {ovnBased ?
                    label(task.num)
                    :
                    requiredInput('num', 'Укажите номер объявления')
                  }
                </Col>
                <Col span={14}>
                  {label(`Дата создания ${task.createdDate}`)}
                </Col>
              </Row>
              <Row>
                <Col span={3}>
                  {label('От кого')}
                </Col>
                <Col span={13}>
                  <Button type="dashed"
                          className={styles.fullwidth}
                          onClick={this.toggleModal}
                  >{task.repFio}</Button>
                  <Modal
                    title="Карточка клиента/представителя клиента"
                    visible={this.state.visible}
                    onCancel={this.handleCancel}
                    cancelText="Отмена"
                    footer={null}
                  >
                    <ClientCard {...representative} />
                  </Modal>
                </Col>
                <Col span={2} className={styles.rightAligned}>
                  <div className={styles.subHead}></div>
                  <div>{label('счет №')}</div>
                </Col>
                <Col span={6}>
                  <div className={styles.subHead}>Дебет</div>
                  {disabledInput('debetAccount')}
                </Col>
              </Row>
              <Row>
                <Col span={3}>
                  {label('Получатель')}
                </Col>
                <Col span={13}>
                  {disabledInput('legalEntityShortName')}
                </Col>
                <Col span={2} className={styles.rightAligned}>
                  <div className={styles.subHead}></div>
                  <div>{label('счет №')}</div>
                </Col>
                <Col span={6}>
                  <div className={styles.subHead}>Кредит</div>
                  {disabledInput('accountId')}
                </Col>
              </Row>
              <Row>
                <Col span={1}>
                  {label('ИНН')}
                </Col>
                <Col span={4}>
                  {disabledInput('inn')}
                </Col>
                <Col span={1}>
                  {label('КПП')}
                </Col>
                <Col span={4}>
                  {disabledInput('kpp')}
                </Col>
                <Col span={2} className={styles.rightAligned}>
                  {label('счет №')}
                </Col>
                <Col span={6}>
                  {disabledInput('accountId')}
                </Col>
                <Col span={6} />
              </Row>
              <Row>
                <Col span={6}>
                  {label('Наименование банка-вносителя')}
                </Col>
                <Col span={12}>
                  {disabledInput('senderBank')}
                </Col>
                <Col span={2} className={styles.rightAligned}>
                  {label('БИК')}
                </Col>
                <Col span={4}>
                  {disabledInput('senderBankBic')}
                </Col>
              </Row>
              <Row>
                <Col span={6}>
                  {label('Наименование банка-получателя')}
                </Col>
                <Col span={12}>
                  {disabledInput('recipientBank')}
                </Col>
                <Col span={2} className={styles.rightAligned}>
                  {label('БИК')}
                </Col>
                <Col span={4}>
                  {disabledInput('recipientBankBic')}
                </Col>
              </Row>
              <Row>
                <Col span={4}>
                  {label('Сумма операции')}
                </Col>
                <Col span={6}>
                  {amountInput()}
                </Col>
              </Row>
              <Row>
                <Col span={4}>
                  {label('Сумма прописью')}
                </Col>
                <Col span={20}>
                  {disabledInput('amountInWords')}
                </Col>
              </Row>
              <Row>
                {label('Источник поступления')}
              </Row>
              <Row>
                {ovnBased ?
                  disabledInput('source')
                  :
                  requiredInput('source', 'Укажите источник поступения')
                }
              </Row>
            </Col>
          </Row>
        </Section>
        { task.clientTypeFk ?
        <Section header="Дополнительная информация">
          <Row>
            {label('Наименование организации')}
          </Row>
          <Row>
            {requiredInput('organisationNameFk', 'Укажите наименование организации')}
          </Row>
          <Row>
            {label('Лицевой счет')}
          </Row>
          <Row>
            {requiredInput('personalAccountId', 'Укажите лицевой счет')}
          </Row>
        </Section>
          : null
        }
      </div>
    )
  }
}
