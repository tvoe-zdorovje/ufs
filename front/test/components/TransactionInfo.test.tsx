import React from 'react'
import {
  render,
  shallow
} from 'enzyme'
import toJson from 'enzyme-to-json'
import { Form } from 'antd'
const FormItem = Form.Item

import TransactionInfo, { TransactionInfoProps } from 'components/CreateNewOperation/TransactionInfo'
import Section from 'components/Section/Section'

const fieldDecoratorMock = () => (component) => component

describe('Компонент <TransactionInfo />', () => {
  const readOnly = false
  const props: TransactionInfoProps = {
    getFieldDecorator: fieldDecoratorMock,
    task: {
      representativeId: '0',
      num: '1',
      createdDate: 'now',
      clientTypeFk: true,
      currencyType: 'RUB',
      accountId: '1',
      legalEntityShortName: '2',
      inn: '3',
      symbols: [],
    },
    representative: {
      inn: '',
      lastName: 'ln',
      firstName: '',
      birthDate: '',
      birthPlace: '',
      resident: true,
      document: {
        type: '',
        series: '',
        number: '',
        issuedBy: '',
        issuedDate: '',
      },
    },
    readOnly,
  }
  const wrapper = shallow(<TransactionInfo {...props} />)
  const renderWrapper = render(<TransactionInfo {...props} />)

  it('использует <Section/> 2 раза', () => {
    expect(wrapper.find(Section)).toHaveLength(2)
  })

  it('отображает компонент <FormItem/> 2 раза', () => {
    expect(wrapper.find(FormItem)).toHaveLength(16)
  })

  it('совпадает со своим снимком', () => {
    expect(toJson(renderWrapper)).toMatchSnapshot()
  })
})
