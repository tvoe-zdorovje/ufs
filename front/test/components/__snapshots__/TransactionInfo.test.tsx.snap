// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент <TransactionInfo /> совпадает со своим снимком 1`] = `
<div
  class="transactionCard"
>
  <section
    class="section"
  >
    <div
      class="header"
    >
      Основная информация
    </div>
    <div
      class="container"
    >
      <div
        class="ant-row"
      >
        <div
          class="ant-col-24"
        >
          <div
            class="ant-row"
          >
            <div
              class="ant-col-10"
            >
              <span
                class="label"
              >
                Объявление №
              </span>
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-form-item-control-wrapper"
                >
                  <div
                    class="ant-form-item-control "
                  >
                    <input
                      class="ant-input ant-input-lg"
                      type="text"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col-14"
            >
              <span
                class="label"
              >
                Дата создания now
              </span>
            </div>
          </div>
          <div
            class="ant-row"
          >
            <div
              class="ant-col-3"
            >
              <span
                class="label"
              >
                От кого
              </span>
            </div>
            <div
              class="ant-col-13"
            >
              <button
                class="ant-btn ant-btn-dashed fullwidth"
                type="button"
              />
            </div>
            <div
              class="ant-col-2 rightAligned"
            >
              <div
                class="subHead"
              />
              <div>
                <span
                  class="label"
                >
                  счет №
                </span>
              </div>
            </div>
            <div
              class="ant-col-6"
            >
              <div
                class="subHead"
              >
                Дебет
              </div>
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-form-item-control-wrapper"
                >
                  <div
                    class="ant-form-item-control "
                  >
                    <input
                      class="ant-input ant-input-lg"
                      disabled=""
                      type="text"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row"
          >
            <div
              class="ant-col-3"
            >
              <span
                class="label"
              >
                Получатель
              </span>
            </div>
            <div
              class="ant-col-13"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-form-item-control-wrapper"
                >
                  <div
                    class="ant-form-item-control "
                  >
                    <input
                      class="ant-input ant-input-lg"
                      disabled=""
                      type="text"
                      value="2"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col-2 rightAligned"
            >
              <div
                class="subHead"
              />
              <div>
                <span
                  class="label"
                >
                  счет №
                </span>
              </div>
            </div>
            <div
              class="ant-col-6"
            >
              <div
                class="subHead"
              >
                Кредит
              </div>
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-form-item-control-wrapper"
                >
                  <div
                    class="ant-form-item-control "
                  >
                    <input
                      class="ant-input ant-input-lg"
                      disabled=""
                      type="text"
                      value="1"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row"
          >
            <div
              class="ant-col-1"
            >
              <span
                class="label"
              >
                ИНН
              </span>
            </div>
            <div
              class="ant-col-4"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-form-item-control-wrapper"
                >
                  <div
                    class="ant-form-item-control "
                  >
                    <input
                      class="ant-input ant-input-lg"
                      disabled=""
                      type="text"
                      value="3"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col-1"
            >
              <span
                class="label"
              >
                КПП
              </span>
            </div>
            <div
              class="ant-col-4"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-form-item-control-wrapper"
                >
                  <div
                    class="ant-form-item-control "
                  >
                    <input
                      class="ant-input ant-input-lg"
                      disabled=""
                      type="text"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col-2 rightAligned"
            >
              <span
                class="label"
              >
                счет №
              </span>
            </div>
            <div
              class="ant-col-6"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-form-item-control-wrapper"
                >
                  <div
                    class="ant-form-item-control "
                  >
                    <input
                      class="ant-input ant-input-lg"
                      disabled=""
                      type="text"
                      value="1"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col-6"
            />
          </div>
          <div
            class="ant-row"
          >
            <div
              class="ant-col-6"
            >
              <span
                class="label"
              >
                Наименование банка-вносителя
              </span>
            </div>
            <div
              class="ant-col-12"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-form-item-control-wrapper"
                >
                  <div
                    class="ant-form-item-control "
                  >
                    <input
                      class="ant-input ant-input-lg"
                      disabled=""
                      type="text"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col-2 rightAligned"
            >
              <span
                class="label"
              >
                БИК
              </span>
            </div>
            <div
              class="ant-col-4"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-form-item-control-wrapper"
                >
                  <div
                    class="ant-form-item-control "
                  >
                    <input
                      class="ant-input ant-input-lg"
                      disabled=""
                      type="text"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row"
          >
            <div
              class="ant-col-6"
            >
              <span
                class="label"
              >
                Наименование банка-получателя
              </span>
            </div>
            <div
              class="ant-col-12"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-form-item-control-wrapper"
                >
                  <div
                    class="ant-form-item-control "
                  >
                    <input
                      class="ant-input ant-input-lg"
                      disabled=""
                      type="text"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-col-2 rightAligned"
            >
              <span
                class="label"
              >
                БИК
              </span>
            </div>
            <div
              class="ant-col-4"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-form-item-control-wrapper"
                >
                  <div
                    class="ant-form-item-control "
                  >
                    <input
                      class="ant-input ant-input-lg"
                      disabled=""
                      type="text"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row"
          >
            <div
              class="ant-col-4"
            >
              <span
                class="label"
              >
                Сумма операции
              </span>
            </div>
            <div
              class="ant-col-6"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-form-item-control-wrapper"
                >
                  <div
                    class="ant-form-item-control "
                  >
                    <span
                      class="ant-input-wrapper ant-input-group"
                    >
                      <input
                        class="ant-input ant-input-lg"
                        type="text"
                      />
                      <span
                        class="ant-input-group-addon"
                      >
                         00 коп.
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row"
          >
            <div
              class="ant-col-4"
            >
              <span
                class="label"
              >
                Сумма прописью
              </span>
            </div>
            <div
              class="ant-col-20"
            >
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-form-item-control-wrapper"
                >
                  <div
                    class="ant-form-item-control "
                  >
                    <input
                      class="ant-input ant-input-lg"
                      disabled=""
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-row"
          >
            <span
              class="label"
            >
              Источник поступления
            </span>
          </div>
          <div
            class="ant-row"
          >
            <div
              class="ant-row ant-form-item"
            >
              <div
                class="ant-form-item-control-wrapper"
              >
                <div
                  class="ant-form-item-control "
                >
                  <input
                    class="ant-input ant-input-lg"
                    type="text"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    class="section"
  >
    <div
      class="header"
    >
      Дополнительная информация
    </div>
    <div
      class="container"
    >
      <div
        class="ant-row"
      >
        <span
          class="label"
        >
          Наименование организации
        </span>
      </div>
      <div
        class="ant-row"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control "
            >
              <input
                class="ant-input ant-input-lg"
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row"
      >
        <span
          class="label"
        >
          Лицевой счет
        </span>
      </div>
      <div
        class="ant-row"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-form-item-control-wrapper"
          >
            <div
              class="ant-form-item-control "
            >
              <input
                class="ant-input ant-input-lg"
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
